README.md

```diff
@@ -2,51 +2,165 @@
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:user_info``` | UserInfo | Yes | The user info structure.
+```:request_id``` | String | Yes | Generated by the SDK (*do not set*)
+```:use_case``` | String | Yes | One of the use case values, i.e. 'FEED' (see [constants.rb](https://github.com/promotedai/promoted-ruby-client/blob/main/lib/promoted/ruby/client/constants.rb)).
+```:properties``` | Properties | Yes | Any additional custom properties to associate.
+```:paging``` | Paging | Yes | Paging parameters (see TODO)
+---
+### MetricsRequest
+Input to ```prepare_for_logging```
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:request``` | Request | No | The underlying request for content.
+```:full_insertion``` | [] of Insertion | No | The proposed list of insertions.
+---
 
-Or
+### DeliveryRequest
+Input to ```deliver```
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:experiment``` | CohortMembership | Yes | A cohort to evaluation in experimentation.
+```:request``` | Request | No | The underlying request for content.
+```:full_insertion``` | [] of Insertion | No | The proposed list of insertions with all metadata, will be compacted before forwarding to Promoted.
+```:only_log``` | Boolean | Yes | Defaults to false. Set to true to override whether Delivery API is called for this request.
+---
 
-1. Clone the repo on your local machine
-2. `cd promoted-ruby-client`
-3. `bundle`
-4. `irb -Ilib -rpromoted/ruby/client`
+### LogRequest
 
-A console will launch with the library loaded.  Here is example code to use.
+Output of ```prepare_for_logging``` as well as an ouput of an SDK call to ```deliver```, input to ```send_log_request``` to log to Promoted
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:request``` | Request | No | The underlying request for content to log.
+```:insertion``` | [] of Insertion | No | The insertions, which are either the original request insertions or the insertions resulting from a call to ```deliver``` if such call occurred.
+---
+
+### ClientResponse
+Output of ```deliver```, includes the insertions as well as a suitable ```LogRequest``` for forwarding to Metrics API.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:insertion``` | [] of Insertion | No | The insertions, which are from Delivery API (when ```deliver``` was called, i.e. we weren't either only-log or part of an experiment) or the input insertions (when the other conditions don't hold).
+```:log_request``` | LogRequest | Yes | A message suitable for logging to Metrics API via ```send_log_request```. If the call to ```deliver``` was made (i.e. the request was not part of the CONTROL arm of an experiment or marked to only log), ```:log_request``` will not be set, as you can assume logging was performed on the server-side by Promoted.
```

Comment:
@danbosnichill I clarified the circumstances under which log_request will be set on the client response, please feel free to double-check.